src/ui/public/field_editor/field_editor.test.js

```diff
@@ -0,0 +1,197 @@
+ * under the License.
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import { FieldEditor } from './field_editor';
+
+jest.mock('@elastic/eui', () => ({
+  EuiButton: 'eui-button',
+  EuiButtonEmpty: 'eui-button-empty',
+  EuiCode: 'eui-code',
+  EuiConfirmModal: 'eui-confirm-modal',
+  EuiFieldNumber: 'eui-field-number',
+  EuiFieldText: 'eui-field-text',
+  EuiFlexGroup: 'eui-flex-group',
+  EuiFlexItem: 'eui-flex-item',
+  EuiForm: 'eui-form',
+  EuiFormRow: 'eui-form-row',
+  EuiIcon: 'eui-icon',
+  EuiLink: 'eui-link',
+  EuiOverlayMask: 'eui-overlay-mask',
+  EuiSelect: 'eui-select',
+  EuiSpacer: 'eui-spacer',
+  EuiText: 'eui-text',
+  EuiTextArea: 'eui-textArea',
+}));
+
+jest.mock('ui/scripting_languages', () => ({
+  GetEnabledScriptingLanguagesProvider: jest.fn().mockImplementation(() => () => ['painless', 'testlang']),
+  getSupportedScriptingLanguages: () => ['painless'],
+  getDeprecatedScriptingLanguages: () => ['testlang'],
+}));
+
+jest.mock('ui/registry/field_formats', () => {
+  class Format {
+    static id = 'test_format'; static title = 'Test format';
+    params() {}
+  }
+
+  return {
+    fieldFormats: {
+      getDefaultType: () => {
+        return Format;
+      },
+      byFieldType: {
+        'number': [Format],
+      },
+    },
+  };
+});
+
+jest.mock('ui/documentation_links', () => ({
+  getDocLink: (doc) => `(docLink for ${doc})`,
+}));
+
+jest.mock('ui/notify', () => ({
+  toastNotifications: {
+    addSuccess: jest.fn(),
+  }
+}));
+
+jest.mock('./components/scripting_call_outs', () => ({
+  ScriptingDisabledCallOut: 'scripting-disabled-callOut',
+  ScriptingWarningCallOut: 'scripting-warning-callOut',
+  ScriptingHelpFlyout: 'scripting-help-flyout',
+}));
+
+jest.mock('./components/field_format_editor', () => ({
+  FieldFormatEditor: 'field-format-editor'
+}));
+
+const fields = [{
+  name: 'foobar',
+}];
+fields.byName = {
+  foobar: {
+    name: 'foobar',
+  },
+};
+
+const indexPattern = {
+  fields,
+};
+
+class Format {
+  static id = 'test_format'; static title = 'Test format';
+  params() {}
+}
+
+const field = {
+  scripted: true,
+  type: 'number',
+  lang: 'painless',
+  format: new Format(),
+};
+
+const helpers = {
+  Field: () => {},
+  getConfig: () => {},
+  $http: () => {},
+  fieldFormatEditors: {},
+  redirectAway: () => {},
+};
+
+describe('FieldEditor', () => {
+  it('should render create new scripted field correctly', async () => {
+    const component = shallow(
+      <FieldEditor
+        indexPattern={indexPattern}
+        field={field}
+        helpers={helpers}
+      />
+    );
+
+    await new Promise(resolve => process.nextTick(resolve));
+    component.update();
+    expect(component).toMatchSnapshot();
+  });
+
+  it('should render edit scripted field correctly', async () => {
+    const testField = {
+      ...field,
+      name: 'test',
+      script: 'doc.test.value',
+    };
+    indexPattern.fields.push(testField);
```

Comment:
Is each test mutating `indexPattern.fields`? If so, then would changing the order of the tests, removing one, or inserting a new one into the middle cause them to break?

Reply:
adjusted tests so that `indexPattern` is initialized in `beforeEach`